sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_uamqp_transport.py

```diff
@@ -658,7 +661,7 @@ def create_receive_client(receiver: "ServiceBusReceiver", **kwargs: Any) -> "Rec
 
         @staticmethod
         def on_attach(  # pylint: disable=unused-argument
-            receiver: "ServiceBusReceiver",
+            receiver: Union["ServiceBusReceiver", "ServiceBusReceiverAsync"],
```

Comment:
do both of these need be under a typing guard instead?

Reply:
Isn't a type guard basically a bool, why would we need that here?

Reply:
yep, I was mistake, was thinking of something else.